ui/src/app/kvm/views/KVMDetails/KVMDetails.tsx

```diff
@@ -34,7 +35,7 @@ export enum KVMAction {
   DELETE = "delete",
   REFRESH = "refresh",
 }
-export type SelectedAction = KVMAction | null;
+export type SelectedAction = KVMAction | MachineAction | null;
```

Comment:
Do you think it would be cleaner to make `SelectedAction` generic and use it for both kvms and machines? It'd mean updating the kvm forms to take that shape, but then you wouldn't have to do the `if (typeof selectedAction === "object"...` check in `KVMActionFormWrapper.tsx`.

Reply:
Yeah I agree we should make `SelectedAction` generic but I think it's going to require a bit of refactoring in both the machine and kvm forms. I'll add an issue to follow up and hopefully have something by the end of the cycle